File: Python/matrixFlipCols.py

def reflectHorizontal (ar: list[list[int]]) -> list[list[int]]:
    for i in range(len(ar)): # number of rows
        left = 0
        right = len(ar[i]) - 1 # number of columns
        while(left < right):
            temp = ar[i][left]
            ar[i][left] = ar[i][right]
            ar[i][right] = temp
            left+=1
            right-=1
    return ar

def printMatrix(ar: list[list[int]]) -> None:
    for i in range(len(ar)):
        print(f'{ar[i]}')

if __name__=='__main__':
    array: list[list[int]] = [
            [0,1,2],
            [1,2,3],
            [2,3,4]
    ]
    string = '-' * 20
    print(f'\nUnreflected Array\n{string}')
    printMatrix(array)
    reflectedArray = reflectHorizontal(array)
    print(f'\nReflected Array\n{string}')
    printMatrix(reflectedArray)

File: go/matrixFlipCols.go
package main

import (
	"fmt"
	"sync"
)

// NOTE: This program flips the rows of a matrix
func main() {
	arr := [][]int{
		{0, 1, 2},
		{1, 2, 3},
		{2, 3, 4},
	}
	str := "-"
	fmt.Printf("\nUnreflected array\n")
	printTwenty := printNTimes(20)
	printTwenty(str)
	printArr(arr)
	reflectHorizontal(arr)
	fmt.Printf("\nReflected array\n")
	printTwenty(str)
	printArr(arr)
}

func printArr(arr [][]int) {
	fmt.Println("-")
	for i := 0; i < len(arr); i++ {
		fmt.Println(arr[i])
	}
}

func reflectHorizontal(ar [][]int) [][]int {
	for i := 0; i < len(ar); i++ { // NOTE: i < number of rows
		left := 0
		right := len(ar[i]) - 1 // NOTE: number of columns
		for left < right {
			temp := ar[i][left]
			ar[i][left] = ar[i][right]
			ar[i][right] = temp
			left++
			right--
		}
	}
	return ar
}

func printNTimes(n int) func(string) {
	f := func(s string) {
		for i := 0; i < n; i++ {
			fmt.Printf("%v", s)
		}
	}
	return f
}
